### ISSUE TITLE
time machine docker image fails for manifests with 'buildtool_export' elements

### ISSUE THREAD

**Issue by gavanderhoorn** [rosin-project/rosinstall_generator_time_machine#17]
Running the time machine with the following command:

```bash
rosinstall_generator_tm.sh 2020-07-21 kinetic --tar --deps roscpp
```

results in the following output:

```
Creating rgtm base dir ..
Need to clone rosdistro .. done
Need to retrieve some rosdistro caches ..done
Requested timepoint: '2020-07-21' (1595282400)
Resetting local rosdistro clone ..
Already on 'master'
Your branch is up to date with 'origin/master'.
Determined rosdistro commit: c6733f65 (authored: 1595333798)
Reverting to ros/rosdistro@c6733f65
Determining closest rosdistro cache ..
Base cache for new cache: /home/user/.robust-rosin/rgtm/rgtm_rosdistro_caches/1541774125
Will store new cache in: /home/user/.robust-rosin/rgtm/rgtm_rosdistro_caches/1595333798
Building cache ..
- trying to use local cache
- trying to fetch cache
- update cache

# skipping all the cache generation output

- write cache file "kinetic-cache.yaml"
- write compressed cache file "kinetic-cache.yaml.gz"
Updating local rosdistro index.yaml to use cache from the past ..
Invoking: rosinstall_generator --rosdistro=kinetic --tar --deps roscpp
Using ROS_DISTRO: kinetic
Traceback (most recent call last):
  File "/rgtm/post141/bin/rosinstall_generator", line 8, in <module>
    sys.exit(main())
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosinstall_generator/cli.py", line 163, in main
    upstream_version_tag=args.upstream, upstream_source_version=args.upstream_development)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosinstall_generator/generator.py", line 364, in generate_rosinstall
    limit_depth=deps_depth, source=upstream_source_version)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosinstall_generator/distro.py", line 91, in get_recursive_dependencies
    ignore_pkgs=dependencies | excludes, limit_depth=limit_depth)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/dependency_walker.py", line 78, in get_recursive_depends
    deps = self.get_depends(next_pkg_to_check, depend_type, ros_packages_only=ros_packages_only)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/dependency_walker.py", line 61, in get_depends
    deps = self._get_dependencies(pkg_name, depend_type)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/dependency_walker.py", line 126, in _get_dependencies
    return set([d.name for d in deps[dep_type]])
KeyError: 'buildtool_export'
```

**Comment by gavanderhoorn** [comment]
It's likely this is caused by the fact we force the use of `rosdistro` version `0.6.8` here:

https://github.com/rosin-project/rosinstall_generator_time_machine/blob/c37a8b6d9f820394d976da155659f8c0075c5132/docker/Dockerfile#L56-L58

ros-infrastructure/rosdistro#134 updated `rosdistro` to support REP-149 (ie: `buildtool_export` etc) and ros-infrastructure/rosinstall_generator#54 depends on it.

As we don't enforce a particular version of `rosinstall_generator` in the `Dockerfile` right now (at least not for the `post141` env, this tries to use an up-to-date version of `rosinstall_generator` with an outdated version of `rosdistro`.

**Comment by gavanderhoorn** [comment]
Might also want to include ros-infrastructure/rosdep#753, so `rosdep` gets smarter as well.

**Comment by gavanderhoorn** [comment]
Related: robust-rosin/robust#268.

**Comment by gavanderhoorn** [comment]
There's a good chance #20 fixes this.

**Comment by gavanderhoorn** [comment]
This should be fixed by #20, so assuming that's the case, I'm closing this.

If it turns out it isn't fixed, we'll re-open.
